src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateManager.ProjectStates.cs

```diff
@@ -67,50 +54,53 @@ public IEnumerable<StateSet> GetAllProjectStateSets()
                 return null;
             }
 
-            private async Task<ProjectAnalyzerStateSets> GetOrCreateProjectStateSetsAsync(Project project, CancellationToken cancellationToken)
-                => TryGetProjectStateSets(project) ?? await UpdateProjectStateSetsAsync(project, cancellationToken).ConfigureAwait(false);
+            private async Task<ProjectAnalyzerInfo> GetOrCreateProjectAnalyzerInfoAsync(Project project, CancellationToken cancellationToken)
+                => TryGetProjectAnalyzerInfo(project) ?? await UpdateProjectAnalyzerInfoAsync(project, cancellationToken).ConfigureAwait(false);
 
-            /// <summary>
-            /// Creates a new project state sets.
-            /// </summary>
-            private ProjectAnalyzerStateSets CreateProjectStateSets(Project project)
+            private ProjectAnalyzerInfo CreateProjectAnalyzerInfo(Project project)
```

Comment:
nit - maybe male a local function inside `UpdateProjectAnalyzerInfoAsync`?

Reply:
yup.  i have a followup that changes a lot of these cases to local functions.